### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
Just some maintainence

### CODE REVIEW

File: src/Editor/GUI/Editors/EntityEditor.cpp [hunk 1]
```diff
@@ -103,31 +103,31 @@ void EntityEditor::Show() {
...
-            value = entity->position.y;
+            value = static_cast<int>(entity->position.y);
             rest = value % toNearest;
 
             if (rest > (toNearest / 2)) {
-                entity->position.y = (value - rest) + toNearest;
+                entity->position.y = static_cast<float>((value - rest) + toNearest);
             } else {
-                entity->position.y = (value - rest);
+                entity->position.y = static_cast<float>(static_cast<float>((value - rest)));
```
Comment (andens) [inline src/Editor/GUI/Editors/EntityEditor.cpp]: What's with the double cast to float?

File: src/Engine/Manager/ParticleManager.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ Component::ParticleSystemComponent* ParticleManager::CreateParticleSystem(const
     particleSystem->particleType.velocity = Json::LoadVec3(node["velocity"]);
     particleSystem->particleType.alpha_control = node.get("alphaControl", 10.0).asFloat();
     particleSystem->particleType.mass = node.get("mass", 1.0).asFloat();
-    particleSystem->particleType.spread = node.get("spread", 1.0).asFloat();
+    particleSystem->particleType.spread = static_cast<int>(node.get("spread", 1.0).asFloat());
```
Comment (andens) [inline src/Engine/Manager/ParticleManager.cpp]: This should probably be something like `.get("spread", 1).asInt();`

File: src/Engine/Util/RayIntersection.cpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ bool RayIntersection::TriangleIntersect(glm::vec3 origin, glm::vec3 direction, g
     glm::vec3 q;
     glm::vec3 r;
     glm::vec3 s;
-    float epsilon = pow(10, -5);
+    float epsilon = static_cast<float>(pow(10, -5));
```
Comment (andens) [inline src/Engine/Util/RayIntersection.cpp]: If I'm not mistaken there might be a `powf` function.

File: src/Video/ParticleSystemRenderer.cpp [hunk 4]
```diff
@@ -18,12 +18,12 @@ ParticleSystemRenderer::ParticleSystemRenderer() {
     srand(time(NULL));
 
     // Temporary rotation matrix for future rotation of particles.
-    rotMat[0][0] = cos(1);
-    rotMat[0][1] = -sin(1);
+    rotMat[0][0] = static_cast<float>(cos(1));
```
Comment (andens) [inline src/Video/ParticleSystemRenderer.cpp]: Maybe there are `sinf` and `cosf` functions for this and the upcoming three occurences.

File: src/Video/ParticleSystemRenderer.cpp [hunk 5]
```diff
@@ -44,12 +44,12 @@ ParticleSystemRenderer::ParticleSystemRenderer(int count) {
     srand(time(NULL));
 
     // Temporary rotation matrix for future rotation of particles.
-    rotMat[0][0] = cos(10);
-    rotMat[0][1] = -sin(10);
+    rotMat[0][0] = static_cast<float>(cos(10));
```
Comment (andens) [inline src/Video/ParticleSystemRenderer.cpp]: See above :)

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Pull request merged**